### PR TITLE
Consistent overriding of labels set by the user

### PR DESCRIPTION
**Pull request by PyvesB**
Hello,

Quite a few badges were ignoring the override label set by the user (using `?label=healthinesses` in the URL for instance), and were just imposing their own value with calls such as `badgeData.text[0] = 'format';`. Admittedly, some of the badges submitted in my previous pull requests were also misbehaving like this! :sweat_smile:

This piece of work makes sure preference is given to the user defined value if any, by adding missing calls to `getLabel` when the value of `badgeData.text[0]` is reassigned.

Everything seems to still pass locally, nevertheless not sure what to put in the pull request title as the list of service tests to run would probably be a bit too long.

Cheers,

Pyves

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -3342,7 +3342,7 @@ cache(function(query_data, match, sendBadge, request) {
           // falls through
         default:
           var value = typeof json_data[info] != 'undefined' && typeof json_data[info] != 'object' ? json_data[info] : Array.isArray(json_data[info]) ? json_data[info].join(", ") : 'invalid data';
-          badgeData.text[0] = query_data.label || type + " " + info;
+          badgeData.text[0] = getLabel(type + " " + info, query_data);
```
Comment (paulmelnikow) [inline server.js]: Could you use a template string?

File: server.js [hunk 2]
```diff
@@ -1784,7 +1784,7 @@ cache(function(queryData, match, sendBadge, request) {
   const variants = {
     // default use `conda|{channelname}` as label
     '': function(queryData, badgeData) {
-      badgeData.text[0] = (queryData && queryData.label) || 'conda|' + badgeData.text[0];
+      badgeData.text[0] = getLabel('conda|' + badgeData.text[0], queryData);
```
Comment (paulmelnikow) [inline server.js]: Could you use a template string? `conda|${badgeData.text[0]}`

File: server.js [hunk 3]
```diff
@@ -6194,21 +6194,21 @@ cache(function(data, match, sendBadge, request) {
...
-        badgeData.text[0] = site + ' reputation';
+        var reputation = parsedData.items[0].reputation;
+        badgeData.text[0] = getLabel(site + ' reputation', data);
         badgeData.text[1] = metric(reputation);
         badgeData.colorscheme = floorCountColor(1000, 10000, 20000);
       } else if (info === 't') {
-        var count = data.items[0].count;
-        badgeData.text[0] = site + ' ' + item + ' questions';
+        var count = parsedData.items[0].count;
+        badgeData.text[0] = getLabel(site + ' ' + item + ' questions', data);
```
Comment (paulmelnikow) [inline server.js]: Template strings?

**Comment by PyvesB** [comment]
Haha, funnily enough I was pair-programming on some Ruby code with a colleague of mine the other day, he suggested something very similar. My main language is Java, there isn't really anything that straightforward built in natively. :wink:

**Comment by paulmelnikow** [comment]
All service tests passing locally.

```
  288 passing (1m)
```

A beautiful thing. 💯

**Pull request merged**